zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/GroupController.java

```diff
@@ -208,12 +227,32 @@ private ResponseEntity<UserSearchResult> searchUsersInGroup(
     }
   }
 
+  private ResponseEntity<RoleSearchQueryResult> searchRolesInGroup(
+      final String groupId, final RoleQuery userQuery) {
+    try {
+      final var composedUserQuery = buildRoleQuery(groupId, userQuery);
+      final var result =
+          roleServices
+              .withAuthentication(RequestMapper.getAuthentication())
+              .search(composedUserQuery);
+      return ResponseEntity.ok(SearchQueryResponseMapper.toRoleSearchQueryResponse(result));
+    } catch (final Exception e) {
+      return mapErrorToResponse(e);
+    }
+  }
+
   private UserQuery buildUserQuery(final String groupId, final UserQuery userQuery) {
     return userQuery.toBuilder()
         .filter(userQuery.filter().toBuilder().groupId(groupId).build())
         .build();
   }
 
+  private RoleQuery buildRoleQuery(final String groupId, final RoleQuery roleQuery) {
+    return roleQuery.toBuilder()
+        .filter(roleQuery.filter().toBuilder().groupId(groupId).build())
+        .build();
```

Comment:
❓ Why do we need to build the filter here, and again in the `DocumentBasedSearchClient`?

Reply:
There is no need to build it twice, I'll remove it from `DocumentBasedSearchClient `